vunit/ui.py

```diff
@@ -447,6 +447,26 @@ def set_generic(self, name, value, allow_empty=False):
         for test_bench in check_not_empty(test_benches, allow_empty, "No test benches found"):
             test_bench.set_generic(name.lower(), value)
 
+    def set_json_generic(self, name, value, allow_empty=False):
```

Comment:
I think it would be better to add a function such as encode_json to make this more orthogonal and avoid having a different variant for each set_generic/set_parameter on the TestBench, Library etc. . Also the add_config method has generic parameter which cannot make use of a set_json_generic.